### PR TITLE
[Hybrid Planning] Address Andy's review

### PR DESCRIPTION
**Pull request by sjahr**
### Description

- Address Andy's review
- Rebase on main

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/hybrid_planning/global_planner/global_planner_plugins/include/moveit/global_planner/moveit_planning_pipeline.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ class MoveItPlanningPipeline : public GlobalPlannerInterface
 {
 public:
   MoveItPlanningPipeline(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/global_planner/global_planner_plugins/include/moveit/global_planner/moveit_planning_pipeline.h:52]: Why not using this:

```suggestion
  MoveItPlanningPipeline() = default;
```

Anyway, regardless of the style semi-colon is redundant:
```suggestion
  MoveItPlanningPipeline(){}
```

File: moveit_ros/hybrid_planning/global_planner/global_planner_plugins/include/moveit/global_planner/moveit_planning_pipeline.h [hunk 2]
```diff
@@ -50,7 +50,7 @@ class MoveItPlanningPipeline : public GlobalPlannerInterface
...
-  ~MoveItPlanningPipeline() override{};
+  ~MoveItPlanningPipeline(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/global_planner/global_planner_plugins/include/moveit/global_planner/moveit_planning_pipeline.h:53]:
```suggestion
  ~MoveItPlanningPipeline(){}
```

File: moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/replan_invalidated_trajectory.h [hunk 3]
```diff
@@ -47,7 +47,7 @@ class ReplanInvalidatedTrajectory : public SinglePlanExecution  // Inherit from
 {
 public:
   ReplanInvalidatedTrajectory(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/replan_invalidated_trajectory.h:49]:
```suggestion
  ReplanInvalidatedTrajectory(){}
```

File: moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/replan_invalidated_trajectory.h [hunk 4]
```diff
@@ -47,7 +47,7 @@ class ReplanInvalidatedTrajectory : public SinglePlanExecution  // Inherit from
...
-  ~ReplanInvalidatedTrajectory() override{};
+  ~ReplanInvalidatedTrajectory(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/replan_invalidated_trajectory.h:50]:
```suggestion
  ~ReplanInvalidatedTrajectory(){}
```

File: moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h [hunk 5]
```diff
@@ -46,7 +46,7 @@ class SinglePlanExecution : public PlannerLogicInterface
 {
 public:
   SinglePlanExecution(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h:48]:
```suggestion
  SinglePlanExecution(){}
```

File: moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h [hunk 6]
```diff
@@ -46,7 +46,7 @@ class SinglePlanExecution : public PlannerLogicInterface
...
-  ~SinglePlanExecution() override{};
+  ~SinglePlanExecution(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h:49]:
```suggestion
  ~SinglePlanExecution(){}
```

File: moveit_ros/hybrid_planning/local_planner/local_constraint_solver_plugins/include/moveit/local_constraint_solver_plugins/forward_trajectory.h [hunk 7]
```diff
@@ -48,7 +48,7 @@ class ForwardTrajectory : public LocalConstraintSolverInterface
 {
 public:
   ForwardTrajectory(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/local_planner/local_constraint_solver_plugins/include/moveit/local_constraint_solver_plugins/forward_trajectory.h:50]:
```suggestion
  ForwardTrajectory(){}
```

File: moveit_ros/hybrid_planning/local_planner/local_constraint_solver_plugins/include/moveit/local_constraint_solver_plugins/forward_trajectory.h [hunk 8]
```diff
@@ -48,7 +48,7 @@ class ForwardTrajectory : public LocalConstraintSolverInterface
...
-  ~ForwardTrajectory() override{};
+  ~ForwardTrajectory(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/local_planner/local_constraint_solver_plugins/include/moveit/local_constraint_solver_plugins/forward_trajectory.h:51]:
```suggestion
  ~ForwardTrajectory(){}
```

File: moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/include/moveit/trajectory_operator_plugins/simple_sampler.h [hunk 9]
```diff
@@ -47,7 +47,7 @@ class SimpleSampler : public TrajectoryOperatorInterface
 {
 public:
   SimpleSampler(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/include/moveit/trajectory_operator_plugins/simple_sampler.h:49]:
```suggestion
  SimpleSampler(){}
```

File: moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/include/moveit/trajectory_operator_plugins/simple_sampler.h [hunk 10]
```diff
@@ -47,7 +47,7 @@ class SimpleSampler : public TrajectoryOperatorInterface
...
-  ~SimpleSampler() override{};
+  ~SimpleSampler(){};
```
Comment (destogl) [inline moveit_ros/hybrid_planning/local_planner/trajectory_operator_plugins/include/moveit/trajectory_operator_plugins/simple_sampler.h:50]:
```suggestion
  ~SimpleSampler(){}
```

**Review by destogl — APPROVED** [review]
Just a few style comments.

**Review by AndyZe — APPROVED** [review]
Approving if you make the changes Denis asked for. Thanks!

**Comment by sjahr** [comment]
> Just a few style comments.
> 
> Can you check and open a PR is somewhere `joint_state_controller` is used instead of `joint_state_broadcaster`. Maybe you have to simply rebase to master.

Done, thanks for your review!

**Comment by sjahr** [comment]
> Approving if you make the changes Denis asked for. Thanks!

Done + I've cleaned up the plugin interface definitions as you suggested

**Comment by AndyZe** [comment]
CI is indicating a failure due to a change in ros2_control, I think:

 ERROR: the following packages/stacks could not have their rosdep keys resolved
  to system dependencies:
  moveit_servo: Cannot locate rosdep definition for [joint_state_controller]
  
  Prob needs to be rebased. I'll merge when it passes CI, especially since it only targets a feature branch.

**Review by henningkayser — APPROVED** [review]

**Pull request merged**